lib/Sema/CSSimplify.cpp

```diff
@@ -4826,6 +4826,27 @@ bool ConstraintSystem::repairFailures(
   case ConstraintLocator::ApplyArgToParam: {
     auto loc = getConstraintLocator(locator);
 
+    // If this type mismatch is associated with a synthesized argument,
+    // let's just ignore it because the main problem is the absence of
+    // the argument.
+    if (auto applyLoc = elt.getAs<LocatorPathElt::ApplyArgToParam>()) {
```

Comment:
@xedin At this point  we are inside of `case ConstraintLocator::ApplyArgToParam:` for `elt.getKind()` which means that elt will be always `LocatorPathElt::ApplyArgToParam` therefore this condition is always true right?